transit.md

```diff
@@ -28,6 +28,36 @@ The other side will attempt to connect to each of those ports, as well as
 listening on its own socket. After a few seconds without success, they will
 both connect to a relay server.
 
+## Transports
+
+The Transit relay supports two kinds of transports: plain TCP streams and WebSockets.
+
+No matter which transport is selected, the same stream of binary data is sent over it.
+That is: first the relay handshake, then the transit handshake and then some number of length-prefixed encrypted records.
+
+### TCP
+
+TCP already provides a stream-oriented protocol, so its handling is straightforward with no extra processing.
+The framing is described in sections below (line-ending based during the handshakes and length-prefixed encrypted records after that).
+
+### WebSockets
+
+The WebSockets protocol is message-based.
+Messages arrive in-order and can be any size up to 4GiB (we recommend using much smaller sizes than this).
+Although the WebSockets protocol handles framing of these messages, when using WebSockets via the Transit relay clients must still regard the payload data as a stream of bytes as outlined above.
+That is, the framing (line-based and then length-prefixed) is still sent inside the WebSockets messages.
+
+All sent and received messages MUST use the WebSockets "binary" scheme.
+Some WebSockets libraries already provide abstractions to treat incoming messages as a stream.
+Stated differently: a single WebSocket message may contain only part of a logical Transit protocol message or it may contain several logical protocol messages; buffering these will be necessary.
+This allows the same protocol parsing to be used for TCP and for WebSockets: simply process all payload bytes in order.
+
+Handling WebSockets in this manner (i.e. instead of ensuring WebSocket messages correspond to individual Transit protocol messages) also allows the Transit relay to be extremely simply, not having to "translate" message framing between the TCP and WebSockets protocols.
+This allows straightforward interoperation between TCP and WebSockets clients with minimal buffering in the Transit server.
+
+More transports may be added in the future.
```

Comment:
Please move that sentence up to the "Transports" section.

Reply:
makes sense, thanks